pkg/controller/transaction/controller.go

```diff
@@ -171,14 +171,31 @@ func (r *Reconciler) reconcileTransactionChangeValidating(ctx context.Context, t
 			return controller.Result{}, errors.NewNotFound("model plugin not found")
 		}
 
+		// validation should be done on the whole config tree if it does exist. the list of path values
+		// that can be used for validation should include both of current path values in the configuration and new ones part of
+		// the transaction
+		configID := configuration.NewID(targetID, change.TargetType, change.TargetVersion)
+		config, err := r.configurations.Get(ctx, configID)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				log.Errorf("Failed applying Transaction %d to target '%s'", transaction.Index, targetID, err)
+				return controller.Result{}, err
+			}
+		}
 		pathValues := make([]*configapi.PathValue, 0, len(change.Values))
+		currentConfigValues := make(map[string]*configapi.PathValue)
+		if config.GetValues() != nil {
+			currentConfigValues = config.Values
+		}
+
 		for path, changeValue := range change.Values {
-			pathValue := &configapi.PathValue{
+			currentConfigValues[path] = &configapi.PathValue{
```

Comment:
I think this should just be called `configValues` because it’s not just current values… it’s the current values plus we’re adding the new ones here.

Reply:
More descriptive might be to call this `pathValues` (since it’s a path to value mapping) and the other one just `values`.